src/cluster_migrateslots.c

```diff
@@ -0,0 +1,1954 @@
+            return;
+        case SLOT_EXPORT_WAITING_TO_SNAPSHOT:
+            /* Perform the snapshot whenever there is no child process.
+             *
+             * We also check that there are no pending writes here. Since we are
+             * sending ACKs at this time, we could have unfortunate timing where
+             * an ACK is added to the output buffer just before snapshotting,
+             * adding the client to the pending write queue. This write might be
+             * flushed after some other command is enqueued during the snapshot,
+             * resulting in premature flush of the output buffer and data
+             * consistency issues. To prevent this, we defer snapshot until
+             * there are no pending writes. */
+            if (hasActiveChildProcess() || job->client->flag.pending_write) {
+                return;
+            }
+            slotExportJobBeginSnapshot(job);
+            return;
+        case SLOT_EXPORT_SNAPSHOTTING:
+            /* Waiting for child process to finish */
+            return;
+        case SLOT_EXPORT_STREAMING:
+            /* Waiting for PAUSE command to come in */
+            return;
+        case SLOT_EXPORT_WAITING_TO_PAUSE:
+            slotExportTryDoPause(job);
+            return;
+        case SLOT_EXPORT_FAILOVER_PAUSED:
+            if (isSlotExportPauseTimedOut(job)) {
+                serverLog(LL_WARNING, "Slot migration %s timed out during slot failover.", job->description);
+                job->mf_end = 0;
+                updatePausedActions();
+                finishSlotMigrationJob(job, SLOT_MIGRATION_JOB_FAILED, "Timed out before streaming completed");
+            }
+            return;
+        case SLOT_EXPORT_FAILOVER_GRANTED:
+            if (isSlotExportPauseTimedOut(job)) {
+                /* Note that we think this won't happen very commonly. The main source of
+                 * latency that may trigger an unpause will occur due to the time it takes
+                 * to process the incremental changes. The final REQUEST-FAILOVER handshake
+                 * will validate that the source node is still paused after this initial
+                 * handshake, and renew the pause for an additional amount of time. From this
+                 * point, we expect the takeover of the slot and gossip to be relatively quick
+                 * in steady state.
+                 *
+                 * Regardless, we log a warning and proceed with cleaning up the slot migration
+                 * job. */
+                serverLog(LL_WARNING, "Write loss risk! During slot migration %s, new owner did not "
+                                      "broadcast ownership before we unpaused ourselves. Any "
+                                      "writes we have recorded since unpausing will now be lost!",
+                          job->description);
+
+                finishSlotMigrationJob(job, SLOT_MIGRATION_JOB_FAILED,
```

Comment:
this failure is fundamentally different than all other failures for the reason you wrote above. how can an admin easily tell the two failure modes apart without parsing the log? can we leave some flags in `CLUSTER GETSLOTMIGRATIONS` output? 

additionally, I think it will be helpful to maintain a counter to track slots ending up in this state.